js/src/sdk/models/connectedAccounts.ts

```diff
@@ -70,16 +79,31 @@ export class ConnectedAccounts {
 
     async initiate(payload: InitiateConnectionDataReq): Promise<ConnectionRequest> {
         try {
-            const {integrationId, entityId = 'default', labels,data={}, redirectUri} = payload;
-      
-            const res = await client.connections.initiateConnection({ body: {
-                integrationId,
-                entityId,
-                labels,
-                redirectUri,
-                data,
-            }  }).then(res => res.data);
-            
+            let { integrationId, entityId = 'default', labels, data = {}, redirectUri, authMode = '', authConfig = {}, appName } = payload;
+
+            if (!integrationId && authMode) {
+                const timestamp = new Date().toISOString().replace(/[-:.]/g, "");
+                const app = await this.apps.get({ appKey: appName! });
+                const integration = await this.integrations.create({
+                    appId: app.appId!,
```

Comment:
Using non-null assertions (`!`) on `app.appId` and `integration.id` could lead to runtime errors if these values are undefined. Consider adding proper validation:
```typescript
if (!app?.appId) {
    throw new Error(`App not found or invalid app ID for ${appName}`);
}
if (!integration?.id) {
    throw new Error('Failed to create integration');
}
integrationId = integration.id;
```

Reply:
I've added it in the next commit